tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -524,9 +585,15 @@ private[optimizer] abstract class OptimizerCore(
           pretransformExpr(expr) { texpr =>
             texpr.tpe match {
               case RefinedType(base: ReferenceType, true, false) =>
+                val actualBase = base match {
+                  case ClassType(LongImpl.RuntimeLongClass) =>
+                    ClassType(Definitions.BoxedLongClass)
+                  case _ =>
+                    base
+                }
```

Comment:
Shouldn't this move to `JSDesugaring`? IIUC, the following will create different results depending on whether the optimizer is on:

```scala
val l = new RuntimeLong
println(l.getClass)
```

I'd feel more comfortable by saying that a `RuntimeLong` is seen as a Long consistently...